sql/expression/function/json_contains_path.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/types"
+)
+
+// JSON_CONTAINS_PATH(json_doc, one_or_all, path[, path] ...)
+//
+// JSONContainsPath Returns 0 or 1 to indicate whether a JSON document contains data at a given path or paths. Returns
+// NULL if any argument is NULL. An error occurs if the json_doc argument is not a valid JSON document, any path
+// argument is not a valid path expression, or one_or_all is not 'one' or 'all'. To check for a specific value at a
+// path, use JSON_CONTAINS() instead.
+//
+// The return value is 0 if no specified path exists within the document. Otherwise, the return value depends on the
```

Comment:
Maybe lead with this? Seems like the important part

Reply:
The NULL behavior seems like a footnote. Not sure JSON_CONTAINS comment needs to be here, but if keeping I'd also put after main explanation.

Reply:
I wasn't going the docs because this is lifted from the link below. I'm happy to re-write it if you think that's better.